src/viking.js

```diff
@@ -1,11 +1,105 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health
+        this.strength = strength
+    };
 
+    attack(){
+        return this.strength
+    };
+
+    receiveDamage(damage){
+        this.health = this.health - damage
+    };
+}
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name,health, strength){
+        super(health, strength);
+        this.name=name;
+    }
+
+
+    receiveDamage(damage) {
+        this.health = this.health - damage
+        if(this.health > 0){
+            return `${this.name} has received ${damage} points of damage`
+        } else { 
+            return `${this.name} has died in act of combat`
+        }
+
+    }
+
+    battleCry(){
+        return "Odin Owns You All!"
+    }
+
+
+
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+    constructor(health, strength){
+        super(health, strength);
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage
+
+        if(this.health > 0){
+            
+            return `A Saxon has received ${damage} points of damage`
+        } else { 
+            return `A Saxon has died in combat`
+        }
+}
+}
+// Bonus War
+ class War {
+    constructor(){
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+       // super(health,strength)
+        //super.receiveDamage();
+        //this.strength=this.strength
```

Comment:
You can remove these commented lines as you are not using them anywhere in your code